### ISSUE TITLE
TensorFlow Profiler did not work correctly.

### ISSUE THREAD

**Issue by y-vectorfield** [tensorflow/tensorflow#62373]
I used the TensorFlow profiler to measure large-scale DL models' CPU/GPU performance. However, this tool did not work correctly. I did not see a graph for the results of measurement using TensorBoard. Hence, I tried a tutorial of [this](https://colab.research.google.com/github/tensorflow/docs-l10n/blob/master/site/ja/tensorboard/tensorboard_profiling_keras.ipynb?hl=ja), and this problem occurred again. I think the current version of the TensorFlow profiler does not save all of the files for measurement results.

The following image shows the current status.
<img width="1986" alt="TensorBoard" src="https://github.com/tensorflow/tensorflow/assets/30323722/b043d495-6999-4383-bac9-6f4fbe255885">

**Comment by tilakrayal** [comment]
@y-vectorfield,
I tried to execute the given tutorial and it was executed without any issue/error. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/c1d0879b01c06230fc2bdf0926cdfe47/tensorboard_profiling_keras.ipynb?hl=ja) and the screenshot for the reference.

![image](https://github.com/tensorflow/tensorflow/assets/81610181/674803e2-cedc-4d5f-82e0-ff166fd3a7e6)


Thank you!

**Comment by TillBeemelmanns** [comment]
Same problem for me. My local setup with TF2.14 produces the same problem "No profile data was found."


If I execute the official notebook https://colab.research.google.com/github/tensorflow/tensorboard/blob/master/docs/tensorboard_profiling_keras.ipynb
then I encounter the same problem.

@tilakrayal What you show in the screenshot is not the profiler !

**Comment by y-vectorfield** [comment]
Hello, @tilakrayal 
@TillBeemelmanns and I could implement TensorBoard.
However, we could not implement TensorFlow Profiler with TensorBoard.
I think this is a problem of the TensorFlow Profiler.

**Comment by akote123** [comment]
Hello @tilakrayal , 
     I wanted to use tensorboard for inference, 
tf.profiler.experimental.start('logs')
output = model.predict(encoded_input, verbose=False)[0]
tf.profiler.experimental.stop()
I have tried this way . But When the profiler tab opened its saying no profile data. May I know how we can get profile data for model inference

**Comment by y-vectorfield** [comment]
@yisitu , @ronshapiro ,
Do you think the above problem?
The TensorFlow Profiler(v2) did not work correctly and did not watch the measurement results.

**Comment by TillBeemelmanns** [comment]
So I could resolve the problem. The solution for me was to manually copy and paste this file https://github.com/tensorflow/profiler/blob/master/demo/events.out.tfevents.1583461681.localhost.profile-empty

into the Tensorboard log directory of my experiment, then suddenly Tensorboard and the profiler-plugin loads the results of the Tensorflow profiler.

**Comment by akote123** [comment]
@TillBeemelmanns ,
       Did you try for inference as well?

**Comment by y-vectorfield** [comment]
@TillBeemelmanns 
Thank you very much for your report.
> The solution for me was to manually copy and paste this file
I think the profiler should create this event file automatically, not manually.

**Comment by sachinprasadhs** [comment]
Hi,

Thanks for reporting the issue, I have reported the issue to the concerned team, I was able to find similar issues which is reported here https://github.com/tensorflow/tensorflow/issues/61212

**Comment by y-vectorfield** [comment]
Hello, @sachinprasadhs 
I confirmed #61212.
This is a similar issue of this.

**Comment by zzzaries** [comment]
Hi y-vectorfield, for the "No data Found" issue, could you please share the screenshot of your log directly after collecting the profile so we can check on the profile data, Thank you!

**Comment by y-vectorfield** [comment]
Hello, @zzzaries, thank you very much for your message.
@akote123 and I tried to measure the inference performance using this.
The example of measurement code is as follows.

```python
with tf.profiler.experimental.Profile(log_dir, options):
    some inference codes are implemented
```

If we implement the above code, we get the following directory structure.

```bash
logs/
└── 20231116-004551
    └── plugins
        └── profile
            └── 2023_11_16_00_46_09
                └── intel-gpu-tensorflow-stable-diffusion-v2140-yvectorfield.xplane.pb
```

We can not get some files of measurement results. Hence, we can not watch the measurement results using TensorBoard.

**Comment by akote123** [comment]
Hi @zzzaries ,
    We also tried with 
tf.profiler.experimental.start('logs')
output = model.predict(encoded_input, verbose=False)[0]

when I do tensorboard --logdir=logs
I get profile page as attached .
<img width="926" alt="tensorboard" src="https://github.com/tensorflow/tensorflow/assets/133775732/ff25c873-e096-4f5f-be6e-34291a592917">

**Comment by y-vectorfield** [comment]
@akote123 , thank you very much for your info sharing.

**Comment by zzzaries** [comment]
Thanks @y-vectorfield @akote123 for the detailed description!

And @akote123 could you also share the directory structure like what y-vectorfield mentioned above? This could be very helpful for us to debug the issue as the directory structure and files inside will potentially determine if tensorboard can identify valid profiler runs data there, Thank you!

**Comment by y-vectorfield** [comment]
@zzzaries, thank you for your message.
Can I help you to fix this issue?
I also try to investigate the root cause of this!

**Comment by akote123** [comment]
@zzzaries , Thank you. the directory structure looks like:
logs/
-plugins
profile/2023_11_17_10_16_20/ip-172-31-93-51.xplane.pb

**Comment by y-vectorfield** [comment]
@akote123, thank you very much for your sharing.

**Comment by akote123** [comment]
@zzzaries , Do you have any example inference code where tensorboard shows the profile data? 
Can you please share any working example for model inference tensorboard code, so that it will be helpful for us to understand whether we are doing any mistake while using profiler API.

Thanks

**Comment by y-vectorfield** [comment]
I used the following code to investigate the root cause of this issue.

```python
from pathlib import Path                                                                                                                                                                                                                                                                                                                                                            

import numpy as np

import tensorflow as tf
from tensorflow.keras.applications.vgg16 import VGG16
from tensorflow.keras.applications.vgg16 import preprocess_input
from tensorflow.keras.preprocessing import image


def main(args):
    model = VGG16(weights='imagenet', include_top=False)
    img_path = 'common/elephant.jpg'
    img = image.load_img(img_path, target_size=(224, 224))
    x = image.img_to_array(img)
    x = np.expand_dims(x, axis=0)
    x = preprocess_input(x)
    log_path = Path(args.log_path)
    if not log_path.exists():
        raise FileNotFoundError(str(log_path))
    with tf.profiler.experimental.Profile(str(log_path)):
        model.predict(x)


if __name__ == '__main__':
    from argparse import ArgumentParser
    arg_parser = ArgumentParser()
    arg_parser.add_argument("--log_path", required=True)
    args = arg_parser.parse_args()
    main(args)
```

**Comment by y-vectorfield** [comment]
I got the following results.

|TF version|eventfile|profiler results|
|:---:|:---:|:---:|
|2.8.0|○|○|
|2.8.1|○|○|
|2.8.2|○|○|
|2.8.4|○|○|
|2.9.0|○|○|
|2.9.1|○|○|
|2.9.2|○|○|
|2.9.3|○|○|
|2.10.0|○|○|
|2.10.1|○|○|
|2.11.0|○|✕|
|2.11.1|○|✕|
|2.12.0|✕|✕|
|2.13.0|✕|✕|
|2.14.0|✕|✕|
|2.15.0|✕|✕|

The result eventfile: ○, profiler results: ○ means the following dir structure.

```bash
logs/
├── events.out.tfevents.1700568471.a0a96e93ba93.profile-empty
└── plugins
    └── profile
        └── 2023_11_21_12_07_51
            ├── a0a96e93ba93.input_pipeline.pb
            ├── a0a96e93ba93.kernel_stats.pb
            ├── a0a96e93ba93.memory_profile.json.gz
            ├── a0a96e93ba93.overview_page.pb
            ├── a0a96e93ba93.tensorflow_stats.pb
            ├── a0a96e93ba93.trace.json.gz
            └── a0a96e93ba93.xplane.pb

3 directories, 8 files
```

The result eventfile: ○, profiler results: ✕ means the following dir structure.

```bash
logs/
├── events.out.tfevents.1700520805.fa34d7d6c817.profile-empty
└── plugins
    └── profile
        └── 2023_11_20_22_53_25
            └── fa34d7d6c817.xplane.pb

3 directories, 2 files
```

The result eventfile: ✕, profiler results: ✕ means the following dir structure.

```bash
logs/
└── plugins
    └── profile
        └── 2023_11_21_05_12_44
            └── d12c6b2d4b00.xplane.pb

3 directories, 1 file
```

**Comment by zzzaries** [comment]
thanks @y-vectorfield  for the detailed debugging and that's really helpful!

for the 3 types of directory structure you showed, the 1 and 2 should both work (2 is intended change), meaning until 2.11 it should be working fine. We've identified some suspicious code snippet and has been working on a fix for this, will update you ASAP.

**Comment by zzzaries** [comment]
> @zzzaries , Do you have any example inference code where tensorboard shows the profile data? Can you please share any working example for model inference tensorboard code, so that it will be helpful for us to understand whether we are doing any mistake while using profiler API.
> 
> Thanks

@akote123 I'm not an expert in performance tuning, but from what I know, you are not doing anything wrong, and this should be a known issue on our side.

However want to double check, as you mentioned 
> We also tried with
> tf.profiler.experimental.start('logs')
> output = model.predict(encoded_input, verbose=False)[0]
> when I do tensorboard --logdir=logs
> I get profile page as attached .

does the above code create the directory of following but you can see the data in the profiler?
and this is all the files within /logs dir?

```
logs/
  -plugins
       -profile
                 -2023_11_17_10_16_20
                                  -ip-172-31-93-51.xplane.pb
```

**Comment by y-vectorfield** [comment]
Hello, @sachinprasadhs, @zzzaries, 
I think we should attach the bug label to
this issue. What do you think my opinion?

**Comment by y-vectorfield** [comment]
@zzzaries,  I would like to contribute to solve this problem. I will investigate the source codes deeply and submit some modification code(Pull Request) as soon as possible. OK?

**Comment by y-vectorfield** [comment]
@sachinprasadhs, thank you!

**Comment by akote123** [comment]
> > @zzzaries , Do you have any example inference code where tensorboard shows the profile data? Can you please share any working example for model inference tensorboard code, so that it will be helpful for us to understand whether we are doing any mistake while using profiler API.
> > Thanks
> 
> @akote123 I'm not an expert in performance tuning, but from what I know, you are not doing anything wrong, and this should be a known issue on our side.
> 
> However want to double check, as you mentioned
> 
> > We also tried with
> > tf.profiler.experimental.start('logs')
> > output = model.predict(encoded_input, verbose=False)[0]
> > when I do tensorboard --logdir=logs
> > I get profile page as attached .
> 
> does the above code create the directory of following but you can see the data in the profiler? and this is all the files within /logs dir?
> 
> ```
> logs/
>   -plugins
>        -profile
>                  -2023_11_17_10_16_20
>                                   -ip-172-31-93-51.xplane.pb
> ```

@zzzaries ,these are the  files created and in the profiler its not showing any data.

**Comment by Laranga22** [comment]
Im expperimenting the same issue I tryed the solution provided by @TillBeemelmanns but this still doesn't work for me

> So I could resolve the problem. The solution for me was to manually copy and paste this file https://github.com/tensorflow/profiler/blob/master/demo/events.out.tfevents.1583461681.localhost.profile-empty
> 
> into the Tensorboard log directory of my experiment, then suddenly Tensorboard and the profiler-plugin load the results of the Tensorflow profiler.

**Comment by y-vectorfield** [comment]
> Im expperimenting the same issue I tryed the solution provided by @TillBeemelmanns but this still doesn't work for me
> 
> > So I could resolve the problem. The solution for me was to manually copy and paste this file https://github.com/tensorflow/profiler/blob/master/demo/events.out.tfevents.1583461681.localhost.profile-empty
> > into the Tensorboard log directory of my experiment, then suddenly Tensorboard and the profiler-plugin load the results of the Tensorflow profiler.

Hello, @Laranga22,
Thank you very much for your comment.
I think the name of *.profile-empty file should match the name of *.pb file.

**Comment by zzzaries** [comment]
> > > @zzzaries , Do you have any example inference code where tensorboard shows the profile data? Can you please share any working example for model inference tensorboard code, so that it will be helpful for us to understand whether we are doing any mistake while using profiler API.
> > > Thanks
> > 
> > 
> > @akote123 I'm not an expert in performance tuning, but from what I know, you are not doing anything wrong, and this should be a known issue on our side.
> > However want to double check, as you mentioned
> > > We also tried with
> > > tf.profiler.experimental.start('logs')
> > > output = model.predict(encoded_input, verbose=False)[0]
> > > when I do tensorboard --logdir=logs
> > > I get profile page as attached .
> > 
> > 
> > does the above code create the directory of following but you can see the data in the profiler? and this is all the files within /logs dir?
> > ```
> > logs/
> >   -plugins
> >        -profile
> >                  -2023_11_17_10_16_20
> >                                   -ip-172-31-93-51.xplane.pb
> > ```
> 
> @zzzaries ,these are the files created and in the profiler its not showing any data.

Thanks akote123 for the detailed info, a file directory like this should display profile data without the tfevetns file (there are different cases due to history changes, while this case with direct `logs/plugins/profile/<run name>/...xplane.pb` should work). Sorry that we are short handed so there's a bit of delay, but we are actively looking into this now.

**Comment by zzzaries** [comment]
Want to double check, for those has the issue of "No Data", do you know what storage system are you using to place the collected profile data? 
or, where are you launching the tensorboard profiler (eg. local, colab notebook, cloud..etc)?

(I've tested with my local device and it's working, but there could be issues for other cases)

**Comment by akote123** [comment]
Hi @zzzaries ,
     Thank you. I am launching tensorboard profiler in AWS machine. 
Could you please share the code you have used to test in local device

**Comment by zzzaries** [comment]
@akote123  is your logdir folder also on the AWS machine where you run the tensorboard profiler?

To test it locally, you can follow instructions on this [quick start](https://github.com/tensorflow/profiler#quick-start)
you could also modify the `install_and_run.py` script (by default it uses the nightly version for tensorflow, tensorboard, and the profiler),  and point `--logdir` to your local  directory to test it out.

`python3 profiler/install_and_run.py --envdir=profile_env --logdir=<path to your local logdir>`

**Comment by zzzaries** [comment]
@akote123 so seems you are able to see the profiler page now, but the data doesn't seem right to you? If so, please file a separate issue in the profiler's github repo at https://github.com/tensorflow/profiler/issues.  It would be nicer to include sample code and steps you use to collect the profile for debugging. Thank you!

**Comment by akote123** [comment]
@zzzaries Thank you . As per your suggestion I have created issue : https://github.com/tensorflow/profiler/issues/784

**Comment by kidneybean-beef** [comment]
> I got the following results. I used each version of TF official container image from Docker Hub. https://hub.docker.com/r/tensorflow/tensorflow/tags?page=1
> 
> TF version	eventfile	profiler results
> 2.8.0	○	○
> 2.8.1	○	○
> 2.8.2	○	○
> 2.8.4	○	○
> 2.9.0	○	○
> 2.9.1	○	○
> 2.9.2	○	○
> 2.9.3	○	○
> 2.10.0	○	○
> 2.10.1	○	○
> 2.11.0	○	✕
> 2.11.1	○	✕
> 2.12.0	✕	✕
> 2.13.0	✕	✕
> 2.14.0	✕	✕
> 2.15.0	✕	✕
> The result eventfile: ○, profiler results: ○ means the following dir structure.
> 
> ```shell
> logs/
> ├── events.out.tfevents.1700568471.a0a96e93ba93.profile-empty
> └── plugins
>     └── profile
>         └── 2023_11_21_12_07_51
>             ├── a0a96e93ba93.input_pipeline.pb
>             ├── a0a96e93ba93.kernel_stats.pb
>             ├── a0a96e93ba93.memory_profile.json.gz
>             ├── a0a96e93ba93.overview_page.pb
>             ├── a0a96e93ba93.tensorflow_stats.pb
>             ├── a0a96e93ba93.trace.json.gz
>             └── a0a96e93ba93.xplane.pb
> 
> 3 directories, 8 files
> ```
> 
> The result eventfile: ○, profiler results: ✕ means the following dir structure.
> 
> ```shell
> logs/
> ├── events.out.tfevents.1700520805.fa34d7d6c817.profile-empty
> └── plugins
>     └── profile
>         └── 2023_11_20_22_53_25
>             └── fa34d7d6c817.xplane.pb
> 
> 3 directories, 2 files
> ```
> 
> The result eventfile: ✕, profiler results: ✕ means the following dir structure.
> 
> ```shell
> logs/
> └── plugins
>     └── profile
>         └── 2023_11_21_05_12_44
>             └── d12c6b2d4b00.xplane.pb
> 
> 3 directories, 1 file
> ```

Hi, have u tried to downgrade the tensorboard to the corresponding tensorflow version? Previously I was using tf 2.5 with the latest tensorboard, and it has this issue. However when I downgraded tensorboard to 2.5 and this "no profile data" issue was solved.

**Comment by MaximeLee** [comment]
> Hi, have u tried to downgrade the tensorboard to the corresponding tensorflow version? Previously I was using tf 2.5 with the latest tensorboard, and it has this issue. However when I downgraded tensorboard to 2.5 and this "no profile data" issue was solved.

I have this problem. But this answer is a hotfix and should not be the final solution.

**Comment by ZM-J** [comment]
Hi, I encountered the same issue recently, and finally I managed to solve it.

Here is my solution, which is also available in my [GitPage](https://zm-j.github.io/2024/04/19/tensorflow-profile-issues/):

> Firstly, for stability, the company's docker images all uniformly use `tensorflow=2.10.0`, even though there are newer versions available now. Therefore, to align with the construction site's setup, we also install this version locally. Tensorboard also needs to be consistent with the tensorflow version (preferably `2.10.0`. Initially, I directly used `pip install tensorflow=2.10` and ended up with `2.10.1`, but later I obediently set it to `2.10.0`)
> 
> Then there's the version of the `tensorboard_plugin_profile` package. Initially, I just followed the official tutorial:
> 
> ```
> pip install tensorboard_plugin_profile
> ```
> 
> to install it, but when it came to launching tensorboard in the browser to view the profile page, I encountered the following content:
> 
> ```md
> **No profile data was found.**
> 
> If you have a model running on CPU, GPU, or Google Cloud TPU, you may be able to use the above button to capture a profile.
> 
> If you're a CPU or GPU user, please use the IP address option. You may want to check out the [tutorial](https://colab.research.google.com/github/tensorflow/tensorboard/blob/master/docs/tensorboard_profiling_keras.ipynb) on how to start a TensorFlow profiler server and profile a Keras model on a GPU.
> 
> If you're a TPU user, please use the TPU name option and you may want to check out the [tutorial](https://cloud.google.com/tpu/docs/cloud-tpu-tools) on how to interpreting the profiling results.
> 
> If you think profiling is done properly, please see the page of [Google Cloud TPU Troubleshooting and FAQ](https://cloud.google.com/tpu/docs/troubleshooting) and consider filing an issue on GitHub.
> ```
> 
> At first, I was completely confused when I saw this, so I went to search and found a few Github issues:
> 
> - <https://github.com/tensorflow/tensorflow/issues/61212>
> - <https://github.com/tensorflow/tensorflow/issues/62373>
> 
> The solutions given there were basically about adjusting the file directory structure, and there was even a monkey patch level fix that involved manually downloading and replacing `events.out.tfevents.TIMESTAMP.USER.profile-empty`. I tried them, but none worked.
> 
> Later, considering that tensorboard is essentially a B-S structure, I checked the server side and found two lines of logs:
> 
> ```
> W tensorflow/core/profiler/convert/xplane_to_tools_data.cc:231] Can not find tool: tool_names. Please update to the latest version of Tensorflow.
> 2024-04-17 14:31:57.295365: W tensorflow/core/profiler/convert/xplane_to_tools_data.cc:231] Can not find tool: tool_names. Please update to the latest version of Tensorflow.
> ```
> 
> Then I speculated that `tool_names` might be a new field introduced in later versions of the tensorflow profiler, but in older versions of tensorboard, this field could not be parsed; and upon refreshing the browser and observing carefully, I noticed there was a loading progress bar, which only declared **No profile data was found** after reaching one-third of the way, which in some way confirmed our speculation. Checking the version of `tensorboard_plugin_profile` package with `pip`, it was found to be `2.15.1`.
> 
> Then I tried installing version `2.10.0`:
> 
> ```
> pip install tensorboard_plugin_profile==2.10.0
> ```
> 
> But still no luck:
> 
> ```
> ERROR: Could not find a version that satisfies the requirement tensorboard_plugin_profile==2.10.0 (from versions: 2.2.0a1, 2.2.0a2, 2.2.0a3, 2.2.0a4, 2.2.0a5, 2.2.0a6, 2.2.0rc0, 2.2.0, 2.3.0rc0, 2.3.0, 2.4.0, 2.5.0, 2.8.0, 2.11.1, 2.11.2, 2.13.0, 2.13.1, 2.14.0, 2.15.0, 2.15.1)
> ERROR: No matching distribution found for tensorboard_plugin_profile==2.10.0
> ```
> 
> I guessed that maybe these plugin-like packages don't get updated as frequently as tensorflow and tensorboard, perhaps the update from `tensorflow=2.8.0` directly jumped to `tensorflow=2.11.1`, so we chose to install the `tensorboard_plugin_profile` version `2.8.0`:
> 
> ```
> pip install tensorboard_plugin_profile==2.8.0
> ```
> 
> After capturing the training records again, tensorboard was able to display the profile information normally.

**Comment by y-vectorfield** [comment]
> Hi, I encountered the same issue recently, and finally I managed to solve it.
> 
> Here is my solution, which is also available in my [GitPage](https://zm-j.github.io/2024/04/19/tensorflow-profile-issues/):
> 
> TL;DR: **Make sure that the version of `tensorboard` & `tensorboard_plugin_profile` is compatible with that of `tensorflow`.**
> 
> > Firstly, for stability, the company's docker images all uniformly use `tensorflow=2.10.0`, even though there are newer versions available now. Therefore, to align with the construction site's setup, we also install this version locally. Tensorboard also needs to be consistent with the tensorflow version (preferably `2.10.0`. Initially, I directly used `pip install tensorflow=2.10` and ended up with `2.10.1`, but later I obediently set it to `2.10.0`)
> > Then there's the version of the `tensorboard_plugin_profile` package. Initially, I just followed the official tutorial:
> > ```
> > pip install tensorboard_plugin_profile
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > to install it, but when it came to launching tensorboard in the browser to view the profile page, I encountered the following content:
> > ```
> > **No profile data was found.**
> > 
> > If you have a model running on CPU, GPU, or Google Cloud TPU, you may be able to use the above button to capture a profile.
> > 
> > If you're a CPU or GPU user, please use the IP address option. You may want to check out the [tutorial](https://colab.research.google.com/github/tensorflow/tensorboard/blob/master/docs/tensorboard_profiling_keras.ipynb) on how to start a TensorFlow profiler server and profile a Keras model on a GPU.
> > 
> > If you're a TPU user, please use the TPU name option and you may want to check out the [tutorial](https://cloud.google.com/tpu/docs/cloud-tpu-tools) on how to interpreting the profiling results.
> > 
> > If you think profiling is done properly, please see the page of [Google Cloud TPU Troubleshooting and FAQ](https://cloud.google.com/tpu/docs/troubleshooting) and consider filing an issue on GitHub.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > At first, I was completely confused when I saw this, so I went to search and found a few Github issues:
> > 
> > * [Tensorflow profiler is not showing anything. Gives "No profile data was found" text on selecting Profile in Tensorboard #61212](https://github.com/tensorflow/tensorflow/issues/61212)
> > * [TensorFlow Profiler did not work correctly. #62373](https://github.com/tensorflow/tensorflow/issues/62373)
> > 
> > The solutions given there were basically about adjusting the file directory structure, and there was even a monkey patch level fix that involved manually downloading and replacing `events.out.tfevents.TIMESTAMP.USER.profile-empty`. I tried them, but none worked.
> > Later, considering that tensorboard is essentially a B-S structure, I checked the server side and found two lines of logs:
> > ```
> > W tensorflow/core/profiler/convert/xplane_to_tools_data.cc:231] Can not find tool: tool_names. Please update to the latest version of Tensorflow.
> > 2024-04-17 14:31:57.295365: W tensorflow/core/profiler/convert/xplane_to_tools_data.cc:231] Can not find tool: tool_names. Please update to the latest version of Tensorflow.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Then I speculated that `tool_names` might be a new field introduced in later versions of the tensorflow profiler, but in older versions of tensorboard, this field could not be parsed; and upon refreshing the browser and observing carefully, I noticed there was a loading progress bar, which only declared **No profile data was found** after reaching one-third of the way, which in some way confirmed our speculation. Checking the version of `tensorboard_plugin_profile` package with `pip`, it was found to be `2.15.1`.
> > Then I tried installing version `2.10.0`:
> > ```
> > pip install tensorboard_plugin_profile==2.10.0
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > But still no luck:
> > ```
> > ERROR: Could not find a version that satisfies the requirement tensorboard_plugin_profile==2.10.0 (from versions: 2.2.0a1, 2.2.0a2, 2.2.0a3, 2.2.0a4, 2.2.0a5, 2.2.0a6, 2.2.0rc0, 2.2.0, 2.3.0rc0, 2.3.0, 2.4.0, 2.5.0, 2.8.0, 2.11.1, 2.11.2, 2.13.0, 2.13.1, 2.14.0, 2.15.0, 2.15.1)
> > ERROR: No matching distribution found for tensorboard_plugin_profile==2.10.0
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I guessed that maybe these plugin-like packages don't get updated as frequently as tensorflow and tensorboard, perhaps the update from `tensorflow=2.8.0` directly jumped to `tensorflow=2.11.1`, so we chose to install the `tensorboard_plugin_profile` version `2.8.0`:
> > ```
> > pip install tensorboard_plugin_profile==2.8.0
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > After capturing the training records again, tensorboard was able to display the profile information normally.

Hello, @ZM-J, thank you very much for your comment. Your information is very valuable. I'd like to refer your experimental results to find out the cause of this problem.

**Comment by tilakrayal** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.

**Comment by y-vectorfield** [comment]
Hello, tilakrayal,
Thank you very mouch for your message.
OK. I'll try the latest version of TensorFlow.

**Comment by Matt-Hurd** [comment]
Hi @tilakrayal and @y-vectorfield. I believe this is an issue with differing implementations between the profiler plugin, and tensorboard for directory identification. I'm currently tracking the issue here: https://github.com/tensorflow/profiler/issues/835. I believe we can keep this issue closed and continue working on it in the profiler repo.

**Comment by y-vectorfield** [comment]
OK,I will check this in a few days.
